### DISCUSSION TITLE
Image background not transparent when paste on another image

### DISCUSSION THREAD

**Discussion by jackthenewbie** [ImageMagick/ImageMagick#5976] [Help] [answered]
### ImageMagick version

7

### Operating system, version and so on

Ubuntu 22.04.1 LTS

### Description

![image](https://user-images.githubusercontent.com/55600733/212523963-339de22c-8895-433c-a57c-e5198c2fa421.png)
Here is a part of my code
```cpp
Image cryo("cryo_dmg_.png");
cryo.alpha(true);
cryo.transparent(Color(cryo.pixelColor(1,1)));
drawList.push_back(DrawableCompositeImage(539, 562+58, 0, 0, cryo));
```
I honestly don't know how to make ``cryo_dmg_.png`` background to be transparent. I tried to looks at the docs, however since the lack of knowledge in image processing i don't know what alpha, matte, etc is. The image perfectly has transparent background when i import to photoshop. As it should be like this:
![image](https://user-images.githubusercontent.com/55600733/212524090-aa713aa4-9338-45d3-a5ca-4cc527884c3b.png)

**Comment by jackthenewbie** [comment, marked as answer]
I dont actually know ``OverCompositeOp`` do (can't understand docs actually), but ye thing works out.
```cpp
Image cryo("cryo_dmg_.png");
drawList.push_back(DrawableCompositeImage(539, 562+58, 0, 0, OverCompositeOp));
```
Note: would be great if someone can link some docs of image processing and common keywords. Thanks.

**Comment by snibgo** [comment]
"Matte" is a word that IM docs once used to mean what is now "alpha".

"Alpha" is a channel sometimes present in images. It usually (but not always) represents opacity, so zero represents no opacity (ie fully transparent) and the maximum value represents full opacity (ie no transparency).

The default "over" composition can be thought of as overlaying sheets of paper. If the top sheet is fully or partly transparent, we can see through it to see the sheet underneath. If the bottom sheet is opaque, we can't see through it to whatever is below it. Placing a transparent sheet *over* an opaque sheet doesn't make the bottom sheet transparent.

Wikipedia articles are useful, eg https://en.wikipedia.org/wiki/Alpha_compositing
